src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,65 @@
 package core.basesyntax;
 
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
 public class WorkWithFile {
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] lines = readFromFile(fromFileName);
+        String report = createReport(lines);
+        writeToFile(report, toFileName);
+    }
+
+    private String[] readFromFile(String fromFileName) {
+
+        File inputFile = new File(fromFileName);
+        Path inputFilePath = Path.of(inputFile.toURI());
+        List<String> lines;
+
+        try {
+            lines = Files.readAllLines(inputFilePath);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read that file named: " + fromFileName, e);
+        }
+        System.out.println(lines);
+        return lines.toArray(new String[]{});
+    }
+
+    private String createReport(String[] lines) {
+        int supplyTotal = 0;
+        int buyTotal = 0;
+
+        for (String line : lines) {
+            String[] parts = line.split(",");
+            String operation = parts[0];
+            int amount = Integer.parseInt(parts[1]);
```

Comment:
You still have magic numbers and string literals here, you should create constants and use them here, for example:

```java
public class WorkWithFile {
    private static final String SEPARATOR = ",";
    private static final int OPERATION_TYPE_INDEX = 0;
    private static final int OPERATION_VALUE_INDEX = 1;
```

And therefore you code should look like this:

```suggestion
            String[] parts = line.split(SEPARATOR);
            String operation = parts[OPERATION_TYPE_INDEX];
            int amount = Integer.parseInt(parts[OPERATION_VALUE_INDEX]);
```